csi.proto

```diff
@@ -271,15 +271,18 @@ message VolumeMetadata {
 }
 
 // A standard way to encode credential data. The total bytes of the values in
-// the Data field must be less than MaxSecretSize bytes.
+// the Data field must be less than 1 Mebibyte.
 message Credentials {
-  // Data contains the credential data, for example username and password. Each
-  // key must consist of alphanumeric characters, '-', '_' or '.'. The
-  // serialized form of the secret data is a base64 encoded string, representing
-  // the arbitrary (possibly non-string) data value here.
-  // This information is sensitive and should be treated as such (not logged,
-  // etc.)
-  map<string, bytes> Data = 1;
+  // Data contains the credential data, for example username and password.
+  // Each key must consist of alphanumeric characters, '-', '_' or '.'.
+  // Each value MUST contain a valid string. An SP MAY choose to accept binary
+  // (non-string) data by using a binary-to-text encoding scheme, like base64.
+  // An SP SHALL advertise the requirements for credentials in documentation.
+  // COs SHALL permit users to pass through the required credentials.
+  // This information is sensitive and MUST be treated as such (not logged,
+  // etc.) by the CO.
+  // This field is REQUIRED.
+  map<string, string> Data = 1;
```

Comment:
> Only if the value contains non-text data will data need to be base64 encoded.

How will the CO know if the value is base64-encoded binary or regular UTF-8 encoded text? Do you perhaps intend that for a given key (eg., "password"), the value must always be UTF-8 encoded text?

I prefer encoding text to UTF-8 on it's way to another process over encoding binary to base64. 

There is always `map<string, oneof[string, bytes]>` (pseudo code)